File: palindrome.py
# def is_palindrome(string: str) -> bool:
#     if not isinstance(string, str) or not string:
#         raise ValueError("Please enter a valid string")

#     reversed_str_list = list(reversed(list(string)))
#     reversed_string = "".join(reversed_str_list)
#     return string.lower().strip() == reversed_string.lower().strip()


def is_palindrome(value: str) -> bool:
    sanitized_value = "".join(char.lower() for char in value if char.isalnum())
    return sanitized_value == sanitized_value[::-1]


print(is_palindrome("eye"))
print(is_palindrome("121"))
print(is_palindrome("Civic"))
print(is_palindrome("Noon"))
print(is_palindrome("DAD"))
print(is_palindrome("Ricardo"))
print(is_palindrome(""))

File: leetcode/palindrome.ts
// const isPalindrome = (
//   value: string,
//   sanitize: (s: string) => string = (s) => s.toLowerCase().replace(/[^a-z0-9]/g, '')
// ): boolean => {
//   const sanitizedValue = sanitize(value);
//   return sanitizedValue === sanitizedValue.split('').reverse().join('');
// };

const isPalindrome = (value: string): boolean => {
  const sanitizedValue = value.toLowerCase().replace(/[^A-z0-9]/g, "")
  return sanitizedValue === sanitizedValue.split("").reverse().join("")
};

console.log(isPalindrome("eye"))
console.log(isPalindrome("121"))
console.log(isPalindrome("Civic"))
console.log(isPalindrome("Noon"))
console.log(isPalindrome("DAD"))
console.log(isPalindrome("Ricardo"))